heft-plugins/heft-storybook-plugin/src/StorybookPlugin.ts

```diff
@@ -267,6 +282,13 @@ export default class StorybookPlugin implements IHeftTaskPlugin<IStorybookPlugin
       ? StorybookBuildMode.WATCH
       : StorybookBuildMode.BUILD;
 
+    if (buildMode === StorybookBuildMode.WATCH && this._isTestMode) {
+      throw new Error('The --storybook-test flag is not supported in watch mode');
+    }
+    if (this._isTestMode && (storybookCliVersion === StorybookCliVersion.STORYBOOK6 || storybookCliVersion === StorybookCliVersion.STORYBOOK7)) {
+      throw new Error('The --storybook-test flag is not supported in storybook6 and storybook7');
```

Comment:
```suggestion
      throw new Error(`The ${storybookTestParameter.longName} flag is only supported in Storybook version 8 and above.`);
```

Reply:
The parameter isn't accessible in this function so I can
1. Duplicate the retrieval of the parameter into this function
2. Store it as a class property like `_isTestMode` so it can be accessed in both places
3. Some alternative

Need to declare `_isTestMode` as it's used in two functions, unless we wanted to pass it in to `_prepareStorybookAsync` and `_runStorybookAsync`

Reply:
Maybe just stick the long name in a const at the top-level of the module.

Reply:
Done. Any thoughts on the addon compatibility "error"? https://github.com/microsoft/rushstack/pull/4971#issuecomment-2412540967

Reply:
Just pushed a minor tweak.

For the compatibility error issue, unfortunately it doesn't look like we can provide a dummy value to `npm_config_user_agent` like "UNSUPPORTED". From that stack trace, and reading through the code in Storybook, just defaulting to npm should work.

Reply:
Have done that. Due to some timing issue with Storybook reading the process environment I was not able to "clean up" the `env` modifications like happens with the `argv` modifications

Reply:
That's probably fine.